--- Mailnag/daemon/mailchecker.py	2013-11-17 17:51:55.560331777 +0100
+++ Mailnag/daemon/mailchecker.py.patched	2013-11-17 17:51:44.853665248 +0100
@@ -22,7 +22,7 @@
 # MA 02110-1301, USA.
 #
 
-from gi.repository import Notify, Gio
+from gi.repository import Notify
 import threading
 import sys
 import subprocess
@@ -45,7 +45,6 @@
 		self._reminder = Reminder()
 		self._pid = Pid()
 		self._cfg = cfg
-		self._gsettings = Gio.Settings.new('org.gnome.shell')
 		# dict that tracks all notifications that need to be closed
 		self._notifications = {}
 		
@@ -100,8 +99,7 @@
 				
 				# play sound if it is enabled in mailnags settings and 
 				# gnome-shell notifications aren't disabled
-				if (self._cfg.get('general', 'playsound') == '1') and \
-					(self._gsettings.get_int('saved-session-presence') != 2):
+				if (self._cfg.get('general', 'playsound') == '1'):
 					gstplay(get_data_file(self._cfg.get('general', 'soundfile')))
 
 			self._reminder.save(self._mail_list)
@@ -147,7 +145,7 @@
 		for mail in new_mails:
 			n = self._get_notification(mail.sender, mail.subject, "mail-unread")
 			notification_id = str(id(n))
-			n.add_action("mark-as-read", _("Mark as read"), self._notification_action_handler, (mail, notification_id), None)			
+			n.add_action("mark-as-read", _("Mark as read"), self._notification_action_handler, (mail, notification_id))			
 			n.show()
 			self._notifications[notification_id] = n
 
@@ -155,7 +153,7 @@
 	def _get_notification(self, summary, body, icon):
 		n = Notify.Notification.new(summary, body, icon)		
 		n.set_category("email")
-		n.add_action("default", "default", self._notification_action_handler, None, None)
+		n.add_action("default", "default", self._notification_action_handler, None)
 
 		return n
 	
